### ISSUE TITLE
ACL: allow access to any resource issue

### ISSUE THREAD

**Issue by iotch** [phalcon/cphalcon#10921]
Hello.
It should be possible to allow access to any resource like that:

``` php
//Allow access to any role to browse on any resource
$acl->allow('*', '*', 'browse');
```

as mentioned in the source - https://github.com/phalcon/cphalcon/blob/master/phalcon/acl/adapter/memory.zep#L437-L438

But it's not possible to allow a wildcard resource, it's forbidden to add an access if resource does not exist:

> Exception 'Phalcon\Acl\Exception' with message 'Access 'browse' does not exist in resource '*''

Attempt to add a wildcard resource leads to:

> Exception 'Phalcon\Acl\Exception' with message 'Resource name cannot be '*''

Thanks.

**Comment by sergeyklay** [comment]
Possible refs #10909

**Comment by iotch** [comment]
I'm not sure.

**Comment by sergeyklay** [comment]
@iotch Can you please check now with 2.0.x branch?

**Comment by stefanchiriac** [comment]
this hasn't been fixed yet? Using phalcon 2.0.10 and still having this problem

**Comment by sergeyklay** [comment]
@stefanchiriac Could you please provide script to reproduce?

**Comment by stefanchiriac** [comment]
Ah, is not exactly the same. Now I see it, but is similar. For example:

    $acl->addResource(new Phalcon\Acl\Resource('*'), $actions);
Error 

     Resource name cannot be '*'
This error makes sense since '*' is a wildcard and shouldn't be a resource. But then if I try directly to allow  all resources and all actions to a role, I get error.

    $acl->allow('admin', '*', '*')
Error

    "Resource '*' does not exist in ACL"
Using Phalcon 2.0.10 from master branch

**Comment by stefanchiriac** [comment]
any chance for this to get in 2.0.x and 2.1.x ?

**Comment by sergeyklay** [comment]
I'll try to deal as soon as I can. Please ping me if I forget

**Comment by stefanchiriac** [comment]
@sergeyklay  any news?

**Comment by stefanchiriac** [comment]
@sergeyklay I see that 2.0.11 is already being prepared for release. So any chance to be fixed in 2.0.11?

**Comment by Jurigag** [comment]
Can be closed according to last comment.
